use-cases/client-server-msgs-perf/svmsg_file_server.c

```diff
@@ -296,33 +319,147 @@ main(int argc, char *argv[])
     }
 #endif  // ! __APPLE__
 
+   // Wait for all threads to complete ...
+   for (int tctr = 0; tctr < num_threads; tctr++) {
+      void *thread_rc;
+      int   rc = pthread_join(thread_ids[tctr], &thread_rc);
+      if (rc) {
+        errExit("pthread_join() failed.");
+      }
+    }
+
+    struct timespec ts1 = {0};
+
+#if !defined(__APPLE__)
+    if (clock_gettime(Clock_id, &ts1)) {        // **** Timing ends
+        errExit("clock_gettime-ts1");
+    }
+#endif  // ! __APPLE__
+
+    assert(NumActiveClients == 0);
+
+    uint64_t nsec0 = timespec_to_ns(&ts0);
+    uint64_t nsec1 = timespec_to_ns(&ts1);
+    uint64_t elapsed_ns = (nsec1 - nsec0);
+
+    if (msgctl(serverId, IPC_RMID, NULL) == -1) {
+        errExit("msgctl serverId");
+    }
+
+    // Close L3-logging upon exit.
+#if defined(L3_LOGT_SPDLOG_BACKTRACE)
+
+    // Log the messages now! show the last n-messages
+    spdlog::dump_backtrace();
+
+#elif L3_ENABLED
+
+    l3_log_deinit(logtype);
+
+#endif  // L3_ENABLED
+
+    printf("Server: # active clients=%d (HWM=%d). Exiting.\n",
+           NumActiveClients, NumActiveClientsHWM);
+
+    printSummaryStats(outfile, run_descr, ActiveClients, NumActiveClientsHWM,
+                      Clock_id, elapsed_ns);
+
+    // For visibility into how clocks are performing on user's machine,
+    // run clock-calibration after all workload / metrics collection is done.
+
+#if defined(__cplusplus)
+
+#if !defined(L3_ENABLED) and !defined(L3_LOGT_SPDLOG) and !defined(L3_LOGT_SPDLOG_BACKTRACE)
+    svr_clock_calibrate();
+#endif
+
+#else   // __cplusplus
+
+#if !defined(L3_ENABLED)
+    svr_clock_calibrate();
+#endif  // L3_ENABLED
+
+#endif  // __cplusplus
+
+    exit(EXIT_SUCCESS);
+}
+
+/**
+ * *****************************************************************************
+ * Server methods to implement each operation.
+ * *****************************************************************************
+ */
+/**
+ * -----------------------------------------------------------------------------
+ * svr_start_threads() - Start 'n' server-threads, receiving and processing
+ * messages from clients.
+ *
+ * Returns:
+ *  0 => Created required # of threads successfully.
+ *  Non-zero => Some error(s) during thread creation.
+ * -----------------------------------------------------------------------------
+ */
+int
+svr_start_threads(pthread_t *thread_ids, int nthreads, void *svr_config)
+{
+    int rv = 0;
+    for (int tctr = 0; tctr < nthreads; tctr++) {
+        rv = pthread_create(&thread_ids[tctr], NULL, svr_proc_process_msg,
+                            svr_config);
+        if (rv) {
+            break;
+        }
+    }
+    return rv;
+}
+
+/**
+ * -----------------------------------------------------------------------------
+ * svr_proc_process_msg() - for()-ever loop to process messages from clients.
+ * -----------------------------------------------------------------------------
+ */
+void *
+svr_proc_process_msg(void *cfg)
```

Comment:
Chunk of for()-ever looping code modularized out from `main()` into this worker function.

Each `case:` block is further modularized out into stand-alone functions. (That may have not been necessary, but I had done that in earlier dev rounds, while trying to create stubs for worker threads to execute.)